AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderBehavior.swift

```diff
@@ -13,4 +13,6 @@ protocol AuthorizationProviderBehavior {
                       completionHandler: @escaping (Result<AuthSession, AuthError>) -> Void)
 
     func invalidateCachedTemporaryCredentials()
+
+    func reset()
```

Comment:
Avoid declaring `reset` directly in behavior protocols in favor of a separate `resettable` conformance on the plugin implementation. Is there a reason we have to put this in the core behavior?

Reply:
Added Resettable conformance to the implementation.